src/Compilers/Core/Portable/Binding/UseSiteInfo.cs

```diff
@@ -82,6 +85,17 @@ public UseSiteInfo<TAssemblySymbol> AdjustDiagnosticInfo(DiagnosticInfo? diagnos
 
             return this;
         }
+
+        public void MergeDependencies(ref TAssemblySymbol? primaryDependency, ref ImmutableHashSet<TAssemblySymbol>? secondaryDependencies)
+        {
+            secondaryDependencies = (secondaryDependencies ?? ImmutableHashSet<TAssemblySymbol>.Empty).Union(SecondaryDependencies ?? ImmutableHashSet<TAssemblySymbol>.Empty);
```

Comment:
Is this the cause of the added Linq using above?

Reply:
If so, this could cause enumerator allocations as it won't use the struct-based enumerator on `ImmutableHashSet`

---
In reply to: [362669134](https://github.com/dotnet/roslyn/pull/40640#discussion_r362669134) [](ancestors = 362669134)

Reply:
> Is this the cause of the added Linq using above?

No, this ```Union``` method is defined on ImmutableHashSet. I assume it should be very efficient when both sides are ImmutableHashSets

---
In reply to: [362669358](https://github.com/dotnet/roslyn/pull/40640#discussion_r362669358) [](ancestors = 362669358,362669134)